x/gov/keeper/hooks_test.go

```diff
@@ -0,0 +1,96 @@
+package keeper_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/gov"
+	"github.com/cosmos/cosmos-sdk/x/gov/keeper"
+	"github.com/cosmos/cosmos-sdk/x/gov/types"
+)
+
+var _ types.GovHooks = &MockGovHooksReceiver{}
+
+// GovHooks event hooks for governance proposal object (noalias)
+type MockGovHooksReceiver struct {
+	AfterProposalSubmissionValid        bool
+	AfterProposalDepositValid           bool
+	AfterProposalVoteValid              bool
+	AfterProposalFailedMinDepositValid  bool
+	AfterProposalVotingPeriodEndedValid bool
+}
+
+func (h *MockGovHooksReceiver) AfterProposalSubmission(ctx sdk.Context, proposalID uint64) {
+	h.AfterProposalSubmissionValid = true
+}
+
+func (h *MockGovHooksReceiver) AfterProposalDeposit(ctx sdk.Context, proposalID uint64, depositorAddr sdk.AccAddress) {
+	h.AfterProposalDepositValid = true
+}
+
+func (h *MockGovHooksReceiver) AfterProposalVote(ctx sdk.Context, proposalID uint64, voterAddr sdk.AccAddress) {
+	h.AfterProposalVoteValid = true
+}
+func (h *MockGovHooksReceiver) AfterProposalFailedMinDeposit(ctx sdk.Context, proposalID uint64) {
+	h.AfterProposalFailedMinDepositValid = true
+}
+func (h *MockGovHooksReceiver) AfterProposalVotingPeriodEnded(ctx sdk.Context, proposalID uint64) {
+	h.AfterProposalVotingPeriodEndedValid = true
+}
+
+func TestHooks(t *testing.T) {
```

Comment:
I wonder if we have a testing suite for this instead?

Reply:
Wdym?